dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -22,12 +22,16 @@ def reset
     # delete the client state and other session state if the user is not signed in
     # and start them at the beginning of the script.
     # If the user is signed in, continue normally.
-    unless current_user
+    redirect_url = build_script_level_path(@script.starting_level)
+
+    if current_user
+      redirect_to(redirect_url) and return
```

Comment:
I think the returns aren't necessary here since this if/else goes to the end of the method